invenio_collections/contrib/search/collection_filter.py

```diff
@@ -17,101 +17,36 @@
     """Create the new AST nodes that should be added to the search query.
 
-    **Set definitions:**
-
-    ``cc``
-        current collection
-
-    ``pr``
-        permitted-restricted collections
-
-    ``r``
-        all restrincted collections
-
-    ``r'``
-        restricted collections for current user (``r' = r - pr``)
-
-    **Policy ANY:**
-
-        (cc AND NOT r') OR (cc and pr) -> cc AND (NOT r' OR rp)
-
-    **Policy ALL:**
-
-        cc AND NOT r'
-
-    :param restricted_cols: all restricted collections
-    :param permitted_restricted_cols: restricted collections that user can
-                                      access
     :param current_col: name of the current collection
-    :param policy: policy applied ('ANY' or enything else)
-    :param format_val: a function used to format value of the boolean
-                       operation
+    :param formatter: function used to generate collection AST
     """
-    current_col_kw = formatter(current_col)
-    result_tree = current_col_kw
+    return formatter(current_col)
 
-    def union_terms(term_list):
-        return reduce(OrOp, [formatter(k) for k in term_list])
 
-    not_permitted_cols = (
-        set(restricted_cols) - set(permitted_restricted_cols)
-    )
-
-    # only if not permitted collection exists
-    if not_permitted_cols:
-        result_tree = NotOp(union_terms(not_permitted_cols))
-
-        if policy == 'ANY' and permitted_restricted_cols:
-            # User needs to have access to at least one collection that
-            # restricts the records. We need this to be able to remove records
-            # that are both in a public and restricted collection.
-
-            result_tree = OrOp(
-                result_tree,
-                union_terms(permitted_restricted_cols)
-            )
-
-        # intersect current collection with restrictions
-        result_tree = AndOp(current_col_kw, result_tree)
-
-    return result_tree
-
-
-def apply(query, user_info=None, collection=None):
+def apply(query, collection=None):
     """Enhance the query restricting not permitted collections.
 
     Get the permitted restricted collection for the current user from the
     user_info object and all the restriced collections from the
     restricted_collection_cache.
     """
-    # FIXME
-    from invenio_collections.cache import restricted_collection_cache
-
-    policy = current_app.config[
-        'COLLECTIONS_VIEW_RESTRICTION_POLICY'].strip().upper()
-    restricted_cols = restricted_collection_cache.cache
-    permitted_restricted_cols = user_info.get(
-        'precached_permitted_restricted_collections', [])
-    current_col = collection or current_app.config['CFG_SITE_NAME']
-    result_tree = create_collection_query(restricted_cols,
-                                          permitted_restricted_cols,
-                                          current_col, policy)
+    current_col = collection or current_app.config['THEME_SITENAME']
```

Comment:
Ah right, I guess it can be the client calling the enhancer that takes care of passing the correct `collection` parameter. If `None` is passed I will return the `query` as is.